### PR TITLE
fix(medusa): update create fulfillment flow

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- Await adjustments made to reservations and inventory items
- use correct map methods and db field

**Why**
- An error thrown during inventory adjustment when creating a fulfillment would not necessarily cause a rollback
- The code in pvinventoryService referenced some older fields for the product_variant_inventory table + used the wrong way to get map values

### CODE REVIEW

**Review by carlos-r-l-rodrigues — APPROVED** [review]
LGTM

non-blocking: We could add an integration test to this flow

File: packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts
Comment (adrien2p) [inline packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts:141]: todo: we should pass the transaction and use it in the function

File: packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts
Comment (adrien2p) [inline packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts:170]: todo: we should use the transaction (related to the previous comment) and create a `const inventoryServiceTx`

File: packages/medusa/src/services/product-variant-inventory.ts
Comment (adrien2p) [inline packages/medusa/src/services/product-variant-inventory.ts:512]: 🙏

**Review by adrien2p — CHANGES_REQUESTED** [review]

File: packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts
Comment (pKorsholm) [reply packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts:141]: we already pass the `Tx` version of the service into the function

File: packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts
Comment (pKorsholm) [reply packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts:170]: same response as above comment 😉

File: packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts
Comment (pKorsholm) [reply packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts:141]:
```
const pvInventoryServiceTx =
        pvInventoryService.withTransaction(transactionManager)

      await updateInventoryAndReservations(
        fulfillments.filter((f) => !existingFulfillmentMap[f.id]),
        {
          inventoryService: pvInventoryServiceTx,
          locationId: validated.location_id,
        }
      )
```

Comment (adrien2p) [reply packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts:139]: My god, i am blind.

File: integration-tests/plugins/__tests__/inventory/order/order.js
Comment (adrien2p) [inline integration-tests/plugins/__tests__/inventory/order/order.js:101]: Question: Deos that mean that the SKU that is used to create the inventory item can be different than the variant SKU? are they meaning different thing?

File: integration-tests/plugins/__tests__/inventory/order/order.js
Comment (adrien2p) [inline integration-tests/plugins/__tests__/inventory/order/order.js:200]: nit: remove the `it`

File: integration-tests/plugins/__tests__/inventory/order/order.js
Comment (adrien2p) [inline integration-tests/plugins/__tests__/inventory/order/order.js:206]: todo: since this id is used a lot, should me move it to a const?

File: integration-tests/plugins/__tests__/inventory/order/order.js
Comment (adrien2p) [inline integration-tests/plugins/__tests__/inventory/order/order.js:232]: todo: rm

File: integration-tests/plugins/__tests__/inventory/order/order.js
Comment (adrien2p) [inline integration-tests/plugins/__tests__/inventory/order/order.js:236]: todo: `err = ....catch(e => e)

File: integration-tests/plugins/__tests__/inventory/order/order.js
Comment (adrien2p) [inline integration-tests/plugins/__tests__/inventory/order/order.js:245]: todo: rm try catch

File: integration-tests/plugins/factories/simple-order-factory.ts
Comment (adrien2p) [inline integration-tests/plugins/factories/simple-order-factory.ts:104]: todo: use isDefined instead

File: integration-tests/plugins/factories/simple-order-factory.ts
Comment (adrien2p) [inline integration-tests/plugins/factories/simple-order-factory.ts:107]: todo: isString

File: integration-tests/plugins/factories/simple-order-factory.ts
Comment (adrien2p) [inline integration-tests/plugins/factories/simple-order-factory.ts:116]: todo: same isString

File: integration-tests/plugins/factories/simple-order-factory.ts
Comment (adrien2p) [inline integration-tests/plugins/factories/simple-order-factory.ts:89]: todo: `Partial<Entity>`

File: integration-tests/plugins/factories/simple-order-factory.ts
Comment (adrien2p) [inline integration-tests/plugins/factories/simple-order-factory.ts:126]: suggestion: With the partial above, you can do `.sales...`

File: packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts
Comment (adrien2p) [inline packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts:125]: suggestion(non blocking): We might take the time to migrate the route to use transformers middleware and cleanup a bit?

File: packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts
Comment (adrien2p) [inline packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts:126]: todo: I can see that we create multiple instance of that service, would it be better to create a const of it?

**Review by adrien2p — CHANGES_REQUESTED** [review]

File: integration-tests/plugins/__tests__/inventory/order/order.js
Comment (pKorsholm) [reply integration-tests/plugins/__tests__/inventory/order/order.js:101]: yes, you will still have access to all the variants stock related fields, however they wont impact behavior if you use the inventory module.

**Comment by olivermrbl** [comment]
@pKorsholm Please resolve conflicts when time allows for it :)

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Comment by olivermrbl** [comment]
Pending @adrien2p's approval :)

File: integration-tests/plugins/factories/simple-order-factory.ts
Comment (adrien2p) [inline integration-tests/plugins/factories/simple-order-factory.ts:129]: todo: if `data.sales_channel`is defined but is not a string, we don't do anything. I am missing it?

**Review by adrien2p — APPROVED** [review]
LGTM, one minore todo if needed

**Review by riqwan — APPROVED** [review]

File: integration-tests/plugins/factories/simple-order-factory.ts
Comment (pKorsholm) [reply integration-tests/plugins/factories/simple-order-factory.ts:129]: if `data.sales_channel` is defined but not a string we pass it into the sales channel factory on line 120

**Review by adrien2p — APPROVED** [review]
LGTM

File: integration-tests/plugins/factories/simple-order-factory.ts
Comment (adrien2p) [reply integration-tests/plugins/factories/simple-order-factory.ts:129]: LOL, I did not see the not in the if constraint. I need to clean my glasses

**Pull request merged**
